homeassistant/components/luxtronik/sensor.py

```diff
@@ -0,0 +1,118 @@
+"""Support for Luxtronik heatpump sensors."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_FRIENDLY_NAME, CONF_ICON, CONF_ID, CONF_SENSORS
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify
+
+from . import DATA_LUXTRONIK, ENTITY_ID_FORMAT
+from .const import (
+    CONF_CALCULATIONS,
+    CONF_GROUP,
+    CONF_PARAMETERS,
+    CONF_VISIBILITIES,
+    DEVICE_CLASSES,
+    ICONS,
+    UNITS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DEVICE_CLASS = "sensor"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_SENSORS): vol.All(
+            cv.ensure_list,
+            [
+                {
+                    vol.Required(CONF_GROUP): vol.All(
+                        cv.string,
+                        vol.Any(CONF_PARAMETERS, CONF_CALCULATIONS, CONF_VISIBILITIES),
+                    ),
+                    vol.Required(CONF_ID): cv.string,
+                    vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+                    vol.Optional(CONF_ICON): cv.string,
+                }
+            ],
+        )
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Luxtronik sensor."""
+    luxtronik = hass.data.get(DATA_LUXTRONIK)
+    if not luxtronik:
+        return False
+
+    sensors = config.get(CONF_SENSORS)
+
+    entities = []
+    for sensor_cfg in sensors:
+        sensor = luxtronik.get_sensor(sensor_cfg[CONF_GROUP], sensor_cfg[CONF_ID])
+        if sensor:
+            entities.append(LuxtronikSensor(luxtronik, sensor, sensor_cfg))
+        else:
+            _LOGGER.warning(
+                "Invalid Luxtronik ID %s in group %s",
+                sensor_cfg[CONF_ID],
+                sensor_cfg[CONF_GROUP],
+            )
+
+    add_entities(entities, True)
+
+
+class LuxtronikSensor(Entity):
+    """Representation of a Luxtronik sensor."""
+
+    def __init__(self, luxtronik, sensor, sensor_cfg):
+        """Initialize a new Luxtronik sensor."""
+        self._luxtronik = luxtronik
+        self._sensor = sensor
+        self._name = sensor_cfg[CONF_FRIENDLY_NAME]
+        self._icon = sensor_cfg[CONF_ICON]
+
+    @property
+    def entity_id(self):
```

Comment:
Remove this.

Reply:
You mean the overriden entity_id, right?
If so, as asked for binary_sensor, is something like [this](https://github.com/home-assistant/core/blob/dev/homeassistant/components/airvisual/sensor.py#L152-L155) the right way to do it?